src/cplscheme/BiCouplingScheme.cpp

```diff
@@ -145,13 +151,17 @@ void BiCouplingScheme::exchangeInitialData()
     }
     if (receivesInitializedData()) {
       receiveData(getM2N(), getReceiveData());
-      checkDataHasBeenReceived();
+    } else {
+      initializeWithZeroInitialData(getReceiveData());
     }
+    checkDataHasBeenReceived();
```

Comment:
This looks odd: Why do we check that data has received if we don't receive it?

Reply:
If we always perform the mapping in `SolverInterfaceImpl::initialize()`, we don't need this change. I think it's indeed better to only `checkDataHasBeenReceived`, if we actually received data. This also requires less changes in the tests. Changed it in a7be0bb7b382cc05241718c9f79ce3cbcd4c21ec.